### ISSUE TITLE
multistore_header appears, even if employees has no right on multi shop

### ISSUE THREAD

**Issue by pppplus** [PrestaShop/PrestaShop#27377]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

I'm on a multishop
I create a employee, with access only to one shop.

On several pages, he see the banner multishop, with all shops available.
He can't access to them, but see their name and url.

### Expected behavior

The multistore_header should not appears if the employee has not right on multishop

### Steps to reproduce

1. Create a multistore
2. Create on employee, with access to only one shop.
3. Connect to this employee account.
4. You will see the multishop header, on all page where it's available (orders, for example)

### PrestaShop version(s) where the bug happened

1.7.8.0

### PHP version(s) where the bug happened

7.4

**Comment by marwachelly** [comment]
Hello,
Thank you for your report. I was able to reproduce the issue with PrestaShop version 1.7.8.0 and develop. 
https://watch.screencastify.com/v/n6tau0V73luwDi2WxGQt
The issue is not reproducible with PS 1.7.7.8 it is a regression.
https://watch.screencastify.com/v/uLaYObAMgnuxwf6VBDqH
 I'll add this to the debug backlog so that it's fixed. Please be aware that due to the high number of reported issues, some bugs might take a very long time to be resolved. If this issue is important to you and you cannot wait for it to be fixed on the project's own time, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you. 

If you manage to get the problem fixed, then please contribute the fix back to the project as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/). Remember that the more people contribute, the better PrestaShop becomes for everyone.

Thanks,

**Comment by atomiix** [comment]
@marwachelly I can't access the second video 🤔
Also, on the first video, we can see that the new user is associated to both shop when you edit its role.

**Comment by marwachelly** [comment]
> @marwachelly I can't access the second video thinking Also, on the first video, we can see that the new user is associated to both shop when you edit its role.

@atomiix  The issue is that the settings of the shop association are not saved. That's lead to wrong right access to the multishop.
PS 1.7.8.0
https://watch.screencastify.com/v/n6tau0V73luwDi2WxGQt

PS 1.7.7.8
https://watch.screencastify.com/v/uLaYObAMgnuxwf6VBDqH

**Comment by pppplus** [comment]
For me employee has ONLY his shop checked. (1.7.8.0)
When I'm connected with his login/password, I see the multistore_header with all stores availables.

Impossible to enter into them, but I see them
![employee](https://user-images.githubusercontent.com/2866100/149959248-0bb96216-1214-4951-902b-a2cc54c474a8.jpg)

employee #7 has only shop 3.

**Comment by atomiix** [comment]
@pppplus you're right, there's 2 bugs here: the first is that shop selection should not be visible when `SuperAdmin` profile is selected and the second is the one you reported!

**Comment by samberrry** [comment]
https://github.com/PrestaShop/PrestaShop/pull/29342

**Comment by samberrry** [comment]
Hello everybody,
this issue fixed by @stifler97 in this PR: https://github.com/PrestaShop/PrestaShop/pull/29342
